src/cephadm/cephadm.py

```diff
@@ -4692,6 +4693,35 @@ def command_list_images(ctx: CephadmContext) -> None:
     # print default images
     cp_obj.write(sys.stdout)
 
+
+def update_service_for_daemon(ctx: CephadmContext, daemon_list: list) -> None:
+    """ Update the unit.meta file of daemon with required service name for valid daemons"""
+
+    data = {'service_name': ctx.service_name}
+    update_daemon = ctx.name.split(',')
+    available_daemons = [d['name'] for d in daemon_list]
+    # check if all the daemon names are invalid
+    if not list(set(update_daemon) & set(available_daemons)):
+        raise Error(f'Error EINVAL: Invalid daemon name {update_daemon}')
+    for name in update_daemon:
+        if name not in available_daemons:
+            print(f'Daemon {name} does not exists')
```

Comment:
IMO we should error out early if any of the daemon names provided aren't present on the host rather than partially completing the operation. The error message in the case where we're actually raising an Error is also a bit misleading to me. The daemon name isn't necessarily invalid (to me that implies formatted incorrectly) but rather that daemon just isn't on the host.

Reply:
Addressed in latest commit